content/_guides/sasl/emacs-erc.md

```diff
@@ -0,0 +1,156 @@
+4. Register: Follow our instructions here to [register a nickname and password](/guides/registration) while signed in.
+5. Once your registration appears successful, then save your username/password to your password safe of preference (Unix *pass*, or *keepassxc*, etc.).
+
+## C. Install ERC-SASL functionality
+1. Download a copy of erc-sasl.el from [Sylvain Benner's GitHub Repo](https://github.com/syl20bnr/spacemacs/blob/master/layers/%2Bchat/erc/local/erc-sasl/erc-sasl.el). (Note-1)
+2. Save it into one of the directorys listed in your emacs’ *load-path* variable. (Note-2)
+3. Add the following to your .emacs or init.el file in your ERC section (Note-3).
+```
+;; Require ERC-SASL package
+(require 'erc-sasl)
+
+;; Add SASL server to list of SASL servers (start a new list, if it did not exist)
+(add-to-list 'erc-sasl-server-regexp-list "irc\\.libera\\.chat")
+
+;; Redefine/Override the erc-login() function from the erc package, so that
+;; it now uses SASL
+(defun erc-login ()
+  "Perform user authentication at the IRC server. (PATCHED)"
+  (erc-log (format "login: nick: %s, user: %s %s %s :%s"
+           (erc-current-nick)
+           (user-login-name)
+           (or erc-system-name (system-name))
+           erc-session-server
+           erc-session-user-full-name))
+  (if erc-session-password
+      (erc-server-send (format "PASS %s" erc-session-password))
+    (message "Logging in without password"))
+  (when (and (featurep 'erc-sasl) (erc-sasl-use-sasl-p))
+    (erc-server-send "CAP REQ :sasl"))
+  (erc-server-send (format "NICK %s" (erc-current-nick)))
+  (erc-server-send
+   (format "USER %s %s %s :%s"
+       ;; hacked - S.B.
+       (if erc-anonymous-login erc-email-userid (user-login-name))
+       "0" "*"
+       erc-session-user-full-name))
+  (erc-update-mode-line))
+```
+4. Change the already-existing connection function you had above, so that it now includes the password you registered
+```
+(defun my-erc-connect ()
+  (interactive)
+  (erc-tls :server "irc.libera.chat" :port 6697 :nick "NICKNAME" :full-name "MY FULL NAME" :password "PASSWORD"))
+(global-set-key (kbd "C-c e") 'my-erc-connect)
+```
+## D. Test
+1. Either close down your emacs completely and restart, or evaluate the lines you added, above, including re-evaluating the my-erc-connect function, and try logging in via SASL, again via your *my-erc-connect* shortcut.
+
+## E. Notes
+- Note-1: If Sylvain’s repo happens to be deleted by the time you read this, then a back-up location is [psachin's gitlab repo](https://gitlab.com/psachin/erc-sasl). The erc-sasl.el there is exactly the same, with one exception: You'll need to change line 54's "loop" to read "cl-loop" (or macro/alias the former to the latter, as some folks do, because cl-loop is a Common Lisp function):
+```
+   (cl-loop for re in erc-sasl-server-regexp-list
+```
+- Note-2: In my case, I saved it to ~/jeff/.emacs.d/elisp/erc-sasl.el because ~/jeff/.emacs.d/elisp/ directory is on my emacs’ *load-path*.
```

Comment:
is Load-path a variable/config setting in emacs? if so it should use backticks rather than **

Reply:
(Yes, it is). Okay, changed, as you recommend.